Add graph problem 3 to CTCI 02/21/2020

iambillzhao opened the pull request:
Problem 3: is it bipartite, the solution is included.
Actions related to this request:
   updated writeup
   updated readme.md
   added solution in Java

Thank you for your review and please let me know if you have any questions.
 -- Bill

etcadinfinitum commented on 2020-01-Winter/5_graphs/README.md:
```diff
@@ -138,21 +164,45 @@ Go to [Top](#top)
+Source: [Leetcode](https://leetcode.com/problems/is-graph-bipartite/)
+
+#### Solution
+
```
I really need a brief explanation of your strategy here in a few sentences. Is it BFS/DFS? What are your terminating conditions? What conditions are you grouping nodes by?

The idea here is that detailing your strategy will help readers understand why you wrote the code that you did, and how it solves the problem you're trying to solve. It would ideally explain the approach the way **you** would explain it if you were facilitating the session on Friday.

etcadinfinitum commented on 2020-01-Winter/5_graphs/README.md:
`````diff
@@ -138,21 +164,45 @@ Go to [Top](#top)
+
+#### Solution
+
+````
`````
Adding the language name here will make sure your code has syntax highlighting, which makes it easier to read.
```suggestion
````java
```

etcadinfinitum commented on 2020-01-Winter/5_graphs/README.md:
`````diff
@@ -67,21 +67,46 @@ Go to [Solution](#s2)   [Top](#top)
 #### Function Signature
 
-TODO :bug:
+````
`````
Add syntax highlighting
```suggestion
````java
```

etcadinfinitum commented on 2020-01-Winter/5_graphs/Bipartite/Bipartite_Solution.java:
```diff
@@ -0,0 +1,31 @@
+ * @author Bill Zhao
+ * @date 02/14/2020
+ */
+
```
It's important to have an executable here that includes a few tests. Could you add that?

iambillzhao commented:
Thank you for your review, Lizzy!
The latest commit** addressed all recommended changes.
Actions related to this request:
added drivers for Bipartite_Solution.java
revised solution description for simplicity
updated syntax highlighting for clarity
** 12bb96d99f69da71875e15692e2ae338bb34343e

Please let me know if you have any questions.
-- Bill

etcadinfinitum commented on 2020-01-Winter/5_graphs/README.md:
`````diff
@@ -67,21 +67,47 @@ Go to [Solution](#s2)   [Top](#top)
 #### Function Signature
 
-TODO :bug:
+```` 
`````
```suggestion
````java
```
